### PR TITLE
Kernel/aarch64: Make the ARM Kernel build with Clang

### PR DESCRIPTION
**Pull request by kleinesfilmroellchen**
Includes a drive-by LibC fix.

### CODE REVIEW

File: Kernel/Arch/aarch64/ASM_wrapper.h [hunk 1]
```diff
@@ -64,11 +64,11 @@ inline void wait_cycles(int n)
 {
```
Comment (FireFox317) [inline Kernel/Arch/aarch64/ASM_wrapper.h:64]: For the commit message, I think you mean: Clang doesn't like using a 32-bit variable with a **64-bit** register
implicitly.

**Review by FireFox317 — APPROVED** [review]
One small comment about the commit message, the rest looks fine to me!

File: Kernel/Arch/aarch64/ASM_wrapper.h [hunk 2]
```diff
@@ -64,11 +64,11 @@ inline void wait_cycles(int n)
...
     // This is probably too fast when caching and branch prediction is turned on.
     // FIXME: Make timer-based.
-    asm("mov x0, %[value]\n"
+    asm("mov w0, %w[value]\n"
         "0:\n"
-        "    subs x0, x0, #1\n"
+        "    subs w0, w0, #1\n"
         "    bne 0b" ::[value] "r"(n)
-        : "x0");
+        : "w0");
```
Comment (Hendiadyoin1) [inline Kernel/Arch/aarch64/ASM_wrapper.h:71]: Even better, you can declare a scratch variable and take it as an in-out, to let the compiler decide which register to use
(might be even possible to write `r` as a clobber and referencing that, but haven't tested)

Comment (kleinesfilmroellchen) [reply Kernel/Arch/aarch64/ASM_wrapper.h:71]: x0 is general purpose in ARM ABIs so idk whether that matters...

**Review by ADKaster — APPROVED** [review]

**Pull request merged**